js/latoken.js

```diff
@@ -0,0 +1,580 @@
+        let response = await this.privateGetOrderActive (this.extend (request, params));
+        let orders = this.parseOrders (response);
+        return orders;
+    }
+
+    async fetchOrder (id, params = {}) {
+        await this.loadMarkets ();
+        let request = {
+            'orderId': id,
+        };
+        let response = await this.privateGetOrderGetOrder (this.extend (request, params));
+        return (this.parseOrder (response));
+    }
+
+    async createOrder (symbol, cliOrdId, side, price, amount, orderType, timeAlive, params = {}) {
+        let order = {
+            'symbol': symbol,
+            'cliOrdId': cliOrdId,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'orderType': orderType,
+            'timeAlive': timeAlive,
+        };
+        let result = await this.privatePostOrderNew (this.extend (order, params));
+        return this.parseOrder (result);
+    }
+
+    async cancelOrder (id, params = {}) {
+        await this.loadMarkets ();
+        if (id === undefined)
+            throw new ArgumentsRequired (this.id + ' cancelOrder requires a id argument');
+        await this.loadMarkets ();
+        let response = await this.privatePostOrderCancel (this.extend ({
+            'orderId': id,
+        }, params));
+        return this.parseOrder (response);
+    }
+
+    async cancelAllOrders (symbol, params = {}) {
+        await this.loadMarkets ();
+        if (symbol === undefined)
+            throw new ArgumentsRequired (this.symbol + ' cancelAllOrders requires a symbol argument');
+        let response = await this.privatePostOrderCancelAll (this.extend ({
+            'symbol': symbol,
+        }, params));
+        let result = [];
+        for (let i = 0; i < response.cancelledOrders.length; i++) {
+            let resp = response.cancelledOrders[i];
+            result.push ({
+                'pairId': response.pairId,
+                'symbol': response.symbol,
+                'cancelledOrder': resp,
+            });
+        }
+        let orders = this.parseTrades (result);
+        return orders;
+    }
+
+    sign (path, api = 'public', method = 'GET', params = undefined, headers = undefined, body = undefined) {
+        let url = this.urls['api']['v1'];
+        url += '/' + path;
+        if (api === 'public') {
+            headers = {
+                'Content-type': 'application/json',
+                'x-lat-timestamp': this.nonce (),
+                'x-lat-timeframe': this.options['timeframe'],
+            };
+            if ((path === 'exchangeInfo/pairs' || 'exchangeInfo/currencies') && (typeof (params) === 'string')) {
+                url += '/' + params;
+            }
+            url += '?' + this.urlencode (params);
+        } else if (api === 'private') {
+            this.checkRequiredCredentials ();
+            if (path === 'account/balances' && (typeof (params) === 'string')) {
```

Comment:
this code here is repeated and params should never be a string - it should be an object.

moreover there are still attribute accesses that use `.attributeName` and these have to be refactored to `x['attributeName']`

Reply:
Code is not  repeated, there is specifity of our API. In three methods params is part of url, not parameters 
Params is object, i fixed
Attribute access i also fixed